server/src/main/java/org/elasticsearch/action/support/ActiveShardCount.java

```diff
@@ -164,19 +164,30 @@ public boolean enoughShardsActive(final ClusterState clusterState, final String.
     }
 
     /**
-     * Returns true iff the active shard count in the shard routing table is enough
-     * to meet the required shard count represented by this instance.
+     * Record that captures the decision of {@link #enoughShardsActive(IndexShardRoutingTable)}.
+     * @param enoughShards the decision of whether the active shard count is enough to meet the required shard count of this instance
+     * @param currentActiveShards the currently active shards considered for making the decision
      */
-    public boolean enoughShardsActive(final IndexShardRoutingTable shardRoutingTable) {
+    public record EnoughShards(boolean enoughShards, int currentActiveShards) {};
+
+    /**
+     * Returns a {@link EnoughShards} record where the first value is true iff the active shard count in the shard routing table is enough
+     * to meet the required shard count represented by this instance, and the second value is the active shard count.
+     */
+    public EnoughShards enoughShardsActive(final IndexShardRoutingTable shardRoutingTable) {
         final int activeShardCount = shardRoutingTable.activeShards().size();
         if (this == ActiveShardCount.ALL) {
-            return activeShardCount == shardRoutingTable.size();
+            return new EnoughShards(activeShardCount == shardRoutingTable.size(), activeShardCount);
         } else if (value == 0) {
-            return true;
+            return new EnoughShards(true, activeShardCount);
         } else if (value == 1) {
-            return shardRoutingTable.hasSearchShards() ? shardRoutingTable.getActiveSearchShardCount() >= 1 : activeShardCount >= 1;
+            if (shardRoutingTable.hasSearchShards()) {
+                return new EnoughShards(shardRoutingTable.getActiveSearchShardCount() >= 1, shardRoutingTable.getActiveSearchShardCount());
+            } else {
+                return new EnoughShards(activeShardCount >= 1, activeShardCount);
+            }
         } else {
-            return shardRoutingTable.getActiveSearchShardCount() >= value;
+            return new EnoughShards(shardRoutingTable.getActiveSearchShardCount() >= value, shardRoutingTable.getActiveSearchShardCount());
```

Comment:
nit: to help with readability, maybe use local variables for activeShardCount and enoughShards and then instanciate he EnoughShard record when returning at the end of the method?